src/Authoring/WinRT.Host/WinRT.Host.cpp

```diff
@@ -43,55 +52,85 @@ static load_assembly_and_get_function_pointer_fn load_assembly_and_get_function_
+        { ResolverResolveFailure, ResolverResolveFailureStr },
+        { LibHostCurExeFindFailure, LibHostCurExeFindFailureStr },
+        { LibHostInitFailure, LibHostInitFailureStr },
+        { LibHostSdkFindFailure, LibHostSdkFindFailureStr },
+        { LibHostInvalidArgs, LibHostInvalidArgsStr },
+        { InvalidConfigFile, InvalidConfigFileStr },
+        { AppArgNotRunnable, AppArgNotRunnableStr },
+        { AppHostExeNotBoundFailure, AppHostExeNotBoundFailureStr },
+        { FrameworkMissingFailure, FrameworkMissingFailureStr },
+        { HostApiFailed, HostApiFailedStr },
+        { HostApiBufferTooSmall, HostApiBufferTooSmallStr },
+        { LibHostUnknownCommand, LibHostUnknownCommandStr },
+        { LibHostAppRootFindFailure, LibHostAppRootFindFailureStr },
+        { SdkResolverResolveFailure, SdkResolverResolveFailureStr },
+        { FrameworkCompatFailure, FrameworkCompatFailureStr },
+        { FrameworkCompatRetry, FrameworkCompatRetryStr },
+        { BundleExtractionFailure, BundleExtractionFailureStr },
+        { BundleExtractionIOError, BundleExtractionIOErrorStr },
+        { LibHostDuplicateProperty, LibHostDuplicatePropertyStr },
+        { HostApiUnsupportedVersion, HostApiUnsupportedVersionStr },
+        { HostInvalidState, HostInvalidStateStr },
+        { HostPropertyNotFound, HostPropertyNotFoundStr },
+        { CoreHostIncompatibleConfig, CoreHostIncompatibleConfigStr },
+        { HostApiUnsupportedScenario, HostApiUnsupportedScenarioStr },
     };
 
+    auto string_table_entry = 0;
     for (auto&& elem : hostfxr_status_messages)
-    {
+    {   
         if (elem.status == result)
         {
-            throw hresult_error(result, elem.message);
+            string_table_entry = elem.resourceID;
+        }
+    }
+
+    if (string_table_entry != 0)
+    {
+        WCHAR error_string[SUFFICIENTLY_LARGE_STRING_BUFFER];
+        WCHAR error_string_buffer[SUFFICIENTLY_LARGE_ERROR_BUFFER];
+
+        HMODULE resources_handle = LoadLibraryExW(WINRT_HOST_FILE_PATH, NULL, LOAD_LIBRARY_AS_IMAGE_RESOURCE | LOAD_LIBRARY_AS_DATAFILE);
+        
+        if (!resources_handle)
+        {
+            // Error if we fail to load WinRT.Host resource file 
+            swprintf_s(error_string_buffer, SUFFICIENTLY_LARGE_ERROR_BUFFER, L"Error: Unable to load the resource container module, last error = %d", GetLastError());
+            MessageBoxW(NULL, error_string_buffer, L"WinRT.Host ERROR!", MB_OK | MB_ICONERROR);
+        } 
+        else if (LoadStringW(resources_handle, string_table_entry, error_string, SUFFICIENTLY_LARGE_STRING_BUFFER) == 0)
+        {
+            // Error if we fail to load the string
+            swprintf_s(error_string_buffer, SUFFICIENTLY_LARGE_ERROR_BUFFER, L"Error: Unable to load the resource string, last error = %d.", GetLastError());
+            MessageBoxW(NULL, error_string_buffer, L"WinRT.Host ERROR!", MB_OK | MB_ICONERROR);
```

Comment:
Should these be message box or should we just throw hresult_error or something else?

Reply:
Thought message box would be better so it's clear what happened instead of error thrown in output. But can change to hresult_error if you rather us stick with that

Reply:
Well the folks who are going to see the error popup is going to be customers of apps which use the authoring component feature.  The app calling this may have its own way of handling errors which the popup can get in the way of and in the current state, it doesn't seem the developer will get crash dumps either due to it isn't an exception.